static/app/components/modals/diffModal.tsx

```diff
@@ -1,17 +1,34 @@
+import {useState} from 'react';
 import {css} from '@emotion/react';
 
 import type {ModalRenderProps} from 'sentry/actionCreators/modal';
+import {Client} from 'sentry/api';
 import IssueDiff from 'sentry/components/issueDiff';
 import useOrganization from 'sentry/utils/useOrganization';
+import {fetchProjectDetails} from 'sentry/views/issueDetails/groupSimilarIssues/similarStackTrace';
 
 type Props = ModalRenderProps & React.ComponentProps<typeof IssueDiff>;
 
 function DiffModal({className, Body, CloseButton, ...props}: Props) {
   const organization = useOrganization();
+  const api = new Client();
+  const {project} = props;
+  const [similarityEmbeddingsProjectFeature, setSimilarityEmbeddingsProjectFeature] =
+    useState<boolean>(false);
+  fetchProjectDetails(api, organization?.slug, project.slug).then(response => {
+    setSimilarityEmbeddingsProjectFeature(
+      response.features.includes('similarity-embeddings')
+    );
+  });
```

Comment:
can you fetch it using this instead? you shouldn't need state afterwards and just check the response features

https://github.com/getsentry/sentry/blob/master/static/app/utils/useDetailedProject.tsx#L18

Reply:
thank you! this is much nicer